openapi_python_client/parser/properties/model_property.py

```diff
@@ -88,10 +90,14 @@ def _check_existing(prop: Property) -> Union[Property, PropertyError]:
     unprocessed_props = data.properties or {}
     for sub_prop in data.allOf or []:
         if isinstance(sub_prop, oai.Reference):
-            source_name = Reference.from_ref(sub_prop.ref).class_name
-            sub_model = schemas.models.get(source_name)
+            ref_path = parse_reference_path(sub_prop.ref)
+            if isinstance(ref_path, ParseError):
+                return PropertyError(detail=ref_path.detail, data=sub_prop)
+            sub_model = schemas.classes_by_reference.get(ref_path)
             if sub_model is None:
                 return PropertyError(f"Reference {sub_prop.ref} not found")
+            if not isinstance(sub_model, ModelProperty):
+                return PropertyError("Cannot take allOf a non-object")
```

Comment:
Is this a new requirement? In our fork, we do things like taking the `allOf` of a string and a string enum.

Reply:
I believe this was already the behavior but it wasn't strictly spelled out. The next couple lines try to iterate through the properties of the model, so if it's not a `ModelProperty` (object) then this fails. There is an open issue #371 to add other types.